packages/framework/src/Component/UploadedFile/UploadedFileFacade.php

```diff
@@ -65,50 +68,82 @@ public function __construct(
 
     /**
      * @param object $entity
-     * @param array|null $temporaryFilenames
+     * @param \Shopsys\FrameworkBundle\Component\UploadedFile\UploadedFileData $uploadedFileData
+     * @param string $type
      */
-    public function uploadFile($entity, $temporaryFilenames)
+    public function manageFiles(object $entity, UploadedFileData $uploadedFileData, string $type = UploadedFileTypeConfig::DEFAULT_TYPE_NAME): void
     {
-        if ($temporaryFilenames !== null && count($temporaryFilenames) > 0) {
-            $entitiesForFlush = [];
-            $uploadedFileEntityConfig = $this->uploadedFileConfig->getUploadedFileEntityConfig($entity);
-            $entityId = $this->getEntityId($entity);
-            $oldUploadedFile = $this->uploadedFileRepository->findUploadedFileByEntity(
-                $uploadedFileEntityConfig->getEntityName(),
-                $entityId
-            );
+        $uploadedFileEntityConfig = $this->uploadedFileConfig->getUploadedFileEntityConfig($entity);
+        $uploadedFileTypeConfig = $uploadedFileEntityConfig->getTypeByName($type);
+
+        $uploadedFiles = $uploadedFileData->uploadedFiles;
+        $orderedFiles = $uploadedFileData->orderedFiles;
 
-            if ($oldUploadedFile !== null) {
-                $this->em->remove($oldUploadedFile);
-                $entitiesForFlush[] = $oldUploadedFile;
+        if ($uploadedFileTypeConfig->isMultiple()) {
```

Comment:
I would like to unify the implementation of this `if` with `ImageFacade`. Logic of this `if` in `ImageFacade` is reversed, it checks if image is not multiple. 
I don't mind which logic should be used.

Reply:
I would like to unify some things too, but I already did a large refactoring and used a lot of time for that. For image refactoring i would need another 5 - 10 hours.

Why I will not change this in UploadedFileFacade is that it is more easier to read and natural to have it this way.